src/k8s/cmd/k8s/k8s_bootstrap_test.go

```diff
@@ -0,0 +1,105 @@
+package k8s
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	. "github.com/onsi/gomega"
+)
+
+var yamlConfigComplete = `
+components:
+  - network
+  - dns
+  - gateway
+  - ingress
+  - storage
+  - metrics-server
+cluster-cidr: "10.244.0.0/16"
+enable-rbac: true
+k8s-dqlite-port: 12379`
+
+var yamlConfigIncomplete = `
+cluster-cidr: "10.244.0.0/16"
+enable-rbac: true
+bananas: 5`
+
+func mustAddConfigToTestDir(t *testing.T, configPath string, data string) {
+	t.Helper()
+	// Create the cluster bootstrap config file
+	err := os.WriteFile(configPath, []byte(data), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetConfigYaml(t *testing.T) {
+	t.Run("CompleteConfig", func(t *testing.T) {
```

Comment:
nit: a [table-driven test](https://golang.cafe/blog/golang-table-test-example) would be less verbose.